### PR TITLE
[RF] testWhenSearchingCityCountry(), testVerifyPopUpWindowShownWhenClickingDifferentWeatherButton, testVerifyChangingTempUnitInHeadingWhenSwitchTempUnitButton<VadimKachanovich>

### PR DESCRIPTION
**Pull request by Vadim-nes**
All tests have been refactored in order to use base methods.
[RF] - https://trello.com/c/gG75rXFa/1155-rf-testwhensearchingcitycountry-testverifypopupwindowshownwhenclickingdifferentweatherbutton-testverifychangingtempunitinheading

### CODE REVIEW

File: src/test/java/VadimKachanovichTest.java [hunk 1]
```diff
@@ -1,84 +1,53 @@
...
-//        click(IMPERIAL_UOM,getWait5());
-//        boolean actualResult = getText(HEADING_UOM,getDriver()).contains("°F");
-//        Assert.assertEquals(actualResult,expectedResult);
-//    }
-//}
+import org.openqa.selenium.By;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import base.BaseTest;
+
```
Comment (larisabratukhina) [inline src/test/java/VadimKachanovichTest.java:5]: delete line, please (next time)

File: src/test/java/VadimKachanovichTest.java [hunk 2]
```diff
@@ -1,84 +1,53 @@
...
+    private final By SEARCH_DROPDOWN_MENU = By.className("search-dropdown-menu");
+    private final By PARIS_FR_DROPDOWN_MENU = By.xpath("//ul[@class = 'search-dropdown-menu']/li/span[text() = 'Paris, FR ']");
+    private final By DIFFERENT_WEATHER_BUTTON = By.xpath("//span[contains(@class,'control-el owm-switch' ) and contains(text(), 'Different Weather?')]");
+    private final By POPUP_MENU_NAME = By.xpath("//div[@class = 'pop-up-container']/div/h3[@id = 'dialogTitle']");
+    private final By IMPERIAL_UOM = By.xpath("//div[@id='weather-widget']//div[@class = 'switch-container']/div[contains(text(),'°F')]");
+    private final By HEADING_UOM = By.xpath("//div[@id='weather-widget']//div[@class = 'section-content']//span[@class='heading']");
+
+    @Test
+    public void testWhenSearchingCityCountry()  {
+
```
Comment (larisabratukhina) [inline src/test/java/VadimKachanovichTest.java:20]: delete line, please (next time)

File: src/test/java/VadimKachanovichTest.java [hunk 3]
```diff
@@ -1,84 +1,53 @@
...
+        final String cityName = "Paris";
+        final String expectedResult = "Paris, FR";
+
+        openBaseURL();
+        String startH2Header = getText(H2_CITY_COUNTRY_HEADER);
```
Comment (larisabratukhina) [inline src/test/java/VadimKachanovichTest.java:25]: add line, please (next time)

File: src/test/java/VadimKachanovichTest.java [hunk 4]
```diff
@@ -1,84 +1,53 @@
...
+        String actualResult = getText(H2_CITY_COUNTRY_HEADER);
+        Assert.assertEquals(actualResult,expectedResult);
+    }
+
+    @Test
+    public void testVerifyPopUpWindowShownWhenClickingDifferentWeatherButton () {
+        final String expectedResult = "Different weather";
+        openBaseURL();
+        click(DIFFERENT_WEATHER_BUTTON);
+        String actualResult = getText(POPUP_MENU_NAME);
```
Comment (larisabratukhina) [inline src/test/java/VadimKachanovichTest.java:41]: add line, please (next time)

File: src/test/java/VadimKachanovichTest.java [hunk 5]
```diff
@@ -1,84 +1,53 @@
...
+        Assert.assertEquals(actualResult,expectedResult);
+    }
+
+    @Test
+    public void testVerifyChangingTempUnitInHeadingWhenSwitchTempUnitButton () {
+        final boolean expectedResult = true;
+        openBaseURL();
+        click(IMPERIAL_UOM);
+        boolean actualResult = getText(HEADING_UOM).contains("°F");
```
Comment (larisabratukhina) [inline src/test/java/VadimKachanovichTest.java:50]: add line, please (next time)

File: src/test/java/VadimKachanovichTest.java [hunk 6]
```diff
@@ -1,84 +1,53 @@
...
+        final String expectedResult = "Different weather";
+        openBaseURL();
+        click(DIFFERENT_WEATHER_BUTTON);
+        String actualResult = getText(POPUP_MENU_NAME);
+        Assert.assertEquals(actualResult,expectedResult);
+    }
+
+    @Test
+    public void testVerifyChangingTempUnitInHeadingWhenSwitchTempUnitButton () {
+        final boolean expectedResult = true;
```
Comment (larisabratukhina) [inline src/test/java/VadimKachanovichTest.java:47]: add line, please (next time)

File: src/test/java/VadimKachanovichTest.java [hunk 7]
```diff
@@ -1,84 +1,53 @@
...
+        openBaseURL();
+        click(IMPERIAL_UOM);
```
Comment (larisabratukhina) [inline src/test/java/VadimKachanovichTest.java:49]: add line, please (next time)

File: src/test/java/VadimKachanovichTest.java [hunk 8]
```diff
@@ -1,84 +1,53 @@
...
+        waitTextToBeChanged(H2_CITY_COUNTRY_HEADER,startH2Header);
+        String actualResult = getText(H2_CITY_COUNTRY_HEADER);
+        Assert.assertEquals(actualResult,expectedResult);
+    }
+
+    @Test
+    public void testVerifyPopUpWindowShownWhenClickingDifferentWeatherButton () {
+        final String expectedResult = "Different weather";
+        openBaseURL();
+        click(DIFFERENT_WEATHER_BUTTON);
```
Comment (larisabratukhina) [inline src/test/java/VadimKachanovichTest.java:40]: add line, please (next time)

File: src/test/java/VadimKachanovichTest.java [hunk 9]
```diff
@@ -1,84 +1,53 @@
...
+        waitElementToBeVisible(SEARCH_DROPDOWN_MENU);
+        click(PARIS_FR_DROPDOWN_MENU);
+        waitTextToBeChanged(H2_CITY_COUNTRY_HEADER,startH2Header);
+        String actualResult = getText(H2_CITY_COUNTRY_HEADER);
+        Assert.assertEquals(actualResult,expectedResult);
+    }
+
+    @Test
+    public void testVerifyPopUpWindowShownWhenClickingDifferentWeatherButton () {
+        final String expectedResult = "Different weather";
```
Comment (larisabratukhina) [inline src/test/java/VadimKachanovichTest.java:38]: add line, please (next time)

File: src/test/java/VadimKachanovichTest.java [hunk 10]
```diff
@@ -1,84 +1,53 @@
...
+    private final By IMPERIAL_UOM = By.xpath("//div[@id='weather-widget']//div[@class = 'switch-container']/div[contains(text(),'°F')]");
+    private final By HEADING_UOM = By.xpath("//div[@id='weather-widget']//div[@class = 'section-content']//span[@class='heading']");
+
+    @Test
+    public void testWhenSearchingCityCountry()  {
+
+        final String cityName = "Paris";
+        final String expectedResult = "Paris, FR";
+
+        openBaseURL();
```
Comment (larisabratukhina) [inline src/test/java/VadimKachanovichTest.java:24]: add line, please (next time)

File: src/test/java/VadimKachanovichTest.java [hunk 11]
```diff
@@ -1,84 +1,53 @@
...
+        String startH2Header = getText(H2_CITY_COUNTRY_HEADER);
+        click(SEARCH_CITY_FIELD);
+        input(cityName,SEARCH_CITY_FIELD);
+        click(SEARCH_BUTTON);
+        waitElementToBeVisible(SEARCH_DROPDOWN_MENU);
+        click(PARIS_FR_DROPDOWN_MENU);
+        waitTextToBeChanged(H2_CITY_COUNTRY_HEADER,startH2Header);
```
Comment (larisabratukhina) [inline src/test/java/VadimKachanovichTest.java:31]: add line, please (next time)

File: src/test/java/VadimKachanovichTest.java [hunk 12]
```diff
@@ -1,84 +1,53 @@
...
+        String actualResult = getText(H2_CITY_COUNTRY_HEADER);
```
Comment (larisabratukhina) [inline src/test/java/VadimKachanovichTest.java:32]: add line, please (next time)

**Review by larisabratukhina — APPROVED** [review]

**Review by verafes — APPROVED** [review]

File: src/test/java/VadimKachanovichTest.java (hunk 12 above)
Comment (Vadim-nes) [reply src/test/java/VadimKachanovichTest.java:32]: Thanks @larisabratukhina. I will do all changes in next PR))

**Pull request merged**